src/plugins/unified_histogram/public/chart/use_stable_callback.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useCallback, useEffect, useRef } from 'react';
+
+/**
+ * Accepts a callback and returns a function with a stable identity
+ * that will always call the latest version of the callback when invoked
+ */
+export const useStableCallback = <T extends (...args: any[]) => any>(fn: T | undefined) => {
```

Comment:
maybe thats's something for `react-use` of @vadimkibana ?

https://github.com/streamich/react-use

Reply:
That's actually where I went looking for an implementation originally, but unfortunately had to fall back on a custom one. Definitely a useful pattern, although I wonder how useful an addition it would be if the React team intends to implement an official version in the near future -- or so they say 😄